### PR TITLE
feat: add documentation on useAzureMonitorSdk feature flag

### PR DESCRIPTION
**Pull request by hkfgo**
Documentation on Azure Monitor SDK migration

Relates to https://github.com/tomkerkhove/promitor/issues/2209

### CODE REVIEW

File: docs/scraping/overview.md [hunk 1]
```diff
@@ -77,6 +77,12 @@ values are `v1`. *(Required)*
 - `azureMetadata.cloud` - The name of the Azure cloud to use. Options are `Global`
  (default), `China`, `UsGov` & `Germany`.
 
+### Azure Monitor Integration
```
Comment (tomkerkhove) [inline docs/scraping/overview.md:80]: Can we move this to https://docs.promitor.io/v2.11/scraping/runtime-configuration/?h=azure+monitor#azure-monitor?

Comment (hkfgo) [reply docs/scraping/overview.md]: Makes sense. I had to go back and make a code change.

**Comment by hkfgo** [comment]
I'm seeing this failure:

```
invalid url - /configuration/v2.x/runtime/scraper/#what-happens-when-metrics-are-unavailable-for-multi-dimensional-metrics [404] [faq.md]
```

Seems unrelated to my change? Let me see if I can fix it..

File: docs/scraping/runtime-configuration.md [hunk 2]
```diff
@@ -351,7 +352,9 @@ The behavior of this can be configured to fit your needs:
  API. (Default: `false`)
 - `azureMonitor.integration.history.startingFromInHours` - Defines the amount of hours Promitor will use to define the starting point of the time window used for metric queries.
   - As an example, the default is 12 hours which means Promitor will fetch all metrics between now - 12 hours and now to find a matching metric. Typically this window can be very small but Promitor provides a margin by default to prevent problems for long aggregation periods. (Default: `12`)
+- `azureMonitor.integration.useAzureMonitorSdk` - In the newest release of Promitor, integration with Azure Monitor will use [Azure.Monitor](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.Query) package under Azure SDK for .NET by default. This migration was needed because the [original SDK](https://github.com/Azure/azure-libraries-for-net) has been deprecated since 2022. 
 
+Since the refactoring may not be completely stable, setting feature flag to false will revert back to using the legacy SDK for scraping.
```
Comment (tomkerkhove) [inline docs/scraping/runtime-configuration.md:357]:
```suggestion
```

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
@hkfgo Can you sign the CLA please?

**Comment by hkfgo** [comment]
Signed!

**Comment by tomkerkhove** [comment]
Thanks!

**Pull request merged**